tests/scripts/audit-validity-dates.py

```diff
@@ -342,20 +347,38 @@ def parse_file(self, filename: str):
                 audit_data = self.parse_bytes(bytes.fromhex(match.group('data')))
                 if audit_data is None:
                     continue
-                audit_data.location = "{}:{}:#{}".format(filename,
-                                                         data_f.line_no,
-                                                         idx + 1)
+                audit_data.locations.append("{}:{}:#{}".format(filename,
+                                                               data_f.line_no,
+                                                               idx + 1))
                 audit_data_list.append(audit_data)
 
         return audit_data_list
 
 
+def merge_auditdata(original: typing.List[AuditData]) \
```

Comment:
Since we're merging identical objects, should the data structure be a `set` rather than a list?

Or should it be a dictionary mapping X.509 objects to a list/set of locations?

Reply:
I'm happy with the changes in 2429113d36027c357309d3bfa0e565e582cf3714, but it seems incomplete, following the [thread about equality and `merge_auditdata`](https://github.com/Mbed-TLS/mbedtls/pull/7512#discussion_r1180528221). Since `AuditData` objects now have a unique identifier, we can use that as a hash and have `merge_auditdata` operate on `set` data structures. Is there any reason not to do that now?

Reply:
Ah, sorry that I forgot to remove `merge_auditdata`. It is not used in current implementation. 

Now we use dictionary to store all the parsed `AuditData` and use the identifier of `AuditData` objects as the key. It's easy for us to maintain a dict with unique `AuditData` objects when we try to add newly parsed `AuditData`. So we don't need `merge_auditdata` anymore, I will remove it.